securedrop_client/api_jobs/uploads.py

```diff
@@ -72,34 +71,44 @@ def call_api(self, api_client: API, session: Session) -> str:
             message = "Failed to send reply for source {id} due to Exception: {error}".format(
                 id=self.source_uuid, error=e)
             raise SendReplyJobTimeoutError(message, self.reply_uuid)
+        except SQLAlchemyError as e:
+            # If the draft reply has already made it to the server but we didn't get a 201 response
+            # back, then a reply with self.reply_uuid will exist in the replies table. Otherwise
+            # the reply was never sent and no longer exists in the drafts table (something went
+            # terribly wrong).
+            try:
+                reply_db_object = session.query(Reply).filter_by(uuid=self.reply_uuid).one()
```

Comment:
This is a WIP PR because I need to make sure catching `SQLAlchemyError` all possble exceptions raised from calling `one`.